src/NServiceBus.Core.Tests/ObjectBuilder/Common/CommonObjectBuilderTests.cs

```diff
@@ -0,0 +1,181 @@
+﻿namespace NServiceBus.Core.Tests
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+    using ObjectBuilder;
+
+    [TestFixture]
+    public class CommonObjectBuilderTests
+    {
+        [SetUp]
+        public void SetUp()
+        {
+            UowDependency.InstanceCounter = 0;
+            InstancePerCallWithUowDependency.InstanceCounter = 0;
+        }
+
+        [Test]
+        public void When_using_injected_builder_to_resolve_uow_dependencies()
```

Comment:
When what? Test name should have a conclusion too. Perhaps `When_using_injected_builder_to_resolve_instancepercalls_with_uow_dependencies_respects_uow_instances`?